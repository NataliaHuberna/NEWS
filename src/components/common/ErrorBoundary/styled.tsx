import styled from "styled-components";
import {COLORS} from '../styles/colors';

export const StMain = styled.div`
  background-image: ${COLORS.PURPLE_BG};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100vh;
`;

export const StContent = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const StBody = styled.div`
  padding: 2.5rem;
  border-radius: 1.5rem;
  background-color: ${COLORS.WHITE_BG};
  max-width: 35rem;
  box-shadow: 0 0.8rem 1.5rem ${COLORS.SHADOW_BOX};
  text-align: center;
`;

export const StTop = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1.5rem;
`;

export const StTitle = styled.h1`
  font-size: 2.3rem;
`;

export const StIcon = styled.i`
  font-size: 2rem;
  margin-left: 1rem;
  position: relative;
  top: 0.1rem;
`;

export const StText = styled.p`
  text-align: left;
  margin-bottom: 2.5rem;
`;

export const StLink = styled.a`
  color: ${COLORS.ORANGE_TEXT};
`;

