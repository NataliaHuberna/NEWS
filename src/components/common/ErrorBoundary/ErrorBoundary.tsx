import React from 'react';
import Footer from '../Footer/Footer';
import {StBody, StContent, StIcon, StLink, StMain, StText, StTitle, StTop} from "./styled";

type TProps = {
    children: React.ReactNode
};

type TState =  {
    error: null ;
};

type PartialPoint = TState | null;
class ErrorBoundary extends React.Component<TProps, PartialPoint> {
    state: TState = {
        error: null
    };

    static getDerivedStateFromError(error: any) {
        return { error };
    }

    render() {
        const { children } = this.props;
        const { error } = this.state;

        if (!error) return children;

        const { name, message } = error;

        return (
            <StMain>
                <StContent>
                    <StBody>
                        <StTop>
                            <StTitle>{name}</StTitle>
                            <StIcon className="fas fa-bug"/>
                        </StTop>

                        <StText>{message}</StText>
                        <StLink href="/">Return to previous page</StLink>
                    </StBody>
                </StContent>

                <Footer/>
            </StMain>
        );
    }
}

export default ErrorBoundary;
