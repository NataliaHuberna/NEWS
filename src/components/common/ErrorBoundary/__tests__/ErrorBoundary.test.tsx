import React from 'react';
import { mount } from 'enzyme';
import ErrorBoundary from "../ErrorBoundary";
import renderer from 'react-test-renderer';

const ComponentWithError = () => { throw new Error();};

describe('ErrorBoundary window',() => {
    it('should match the snapshot', () => {
        const tree = renderer.create(<ErrorBoundary><ComponentWithError/></ErrorBoundary> ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('displays error message on error generated by child', () => {
        const wrapper = mount(
            <ErrorBoundary >
                <ComponentWithError />
            </ErrorBoundary>
        );
        expect(wrapper.find('h1').text()).toEqual('Error');
    });
});
