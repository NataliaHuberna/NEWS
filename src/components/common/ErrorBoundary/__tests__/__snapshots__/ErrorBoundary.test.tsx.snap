// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary> window should match the snapshot 1`] = `
<div
  className="sc-hKwDye eZOoiD"
>
  <div
    className="sc-eCImPb jMSvpK"
  >
    <div
      className="sc-jRQBWg bACNkJ"
    >
      <div
        className="sc-gKclnd fZvJuZ"
      >
        <h1
          className="sc-iCfMLu dLDEDg"
        >
          Error
        </h1>
        <i
          className="sc-furwcr eVOYR fas fa-bug"
        />
      </div>
      <p
        className="sc-pVTFL fPCuWM"
      >
        
      </p>
      <a
        className="sc-jrQzAO fsxzBU"
        href="/aboutUs"
      >
        Return to main page
      </a>
    </div>
  </div>
  <div
    className="sc-bdvvtL cjbcZU"
  >
    <div
      className="sc-gsDKAQ gpLSln"
    >
      <strong
        className="sc-dkPtRN eHRfMh"
      >
        All Rights Reserved, 
        2022
      </strong>
    </div>
  </div>
</div>
`;

exports[`ErrorBoundary window should match the snapshot 1`] = `
<div
  className="sc-hKwDye eZOoiD"
>
  <div
    className="sc-eCImPb jMSvpK"
  >
    <div
      className="sc-jRQBWg bACNkJ"
    >
      <div
        className="sc-gKclnd fZvJuZ"
      >
        <h1
          className="sc-iCfMLu dLDEDg"
        >
          Error
        </h1>
        <i
          className="sc-furwcr eVOYR fas fa-bug"
        />
      </div>
      <p
        className="sc-pVTFL fPCuWM"
      >
        
      </p>
      <a
        className="sc-jrQzAO fsxzBU"
        href="/aboutUs"
      >
        Return to main page
      </a>
    </div>
  </div>
  <div
    className="sc-bdvvtL cjbcZU"
  >
    <div
      className="sc-gsDKAQ gpLSln"
    >
      <strong
        className="sc-dkPtRN eHRfMh"
      >
        All Rights Reserved, 
        2022
      </strong>
    </div>
  </div>
</div>
`;
